client/pdf_generator.py

```diff
@@ -0,0 +1,186 @@
+	options = [
+		plugins.ClientOptionString(
+			'output_pdf',
+			'pdf being generated',
+			default='~/Attachment1.pdf',
+			display_name='* Output PDF File'
+		),
+		plugins.ClientOptionString(
+			'template_file',
+			'Template file to read from',
+			default='~/template.txt',
+			display_name='* Template File'
+		),
+		plugins.ClientOptionString(
+			'logo',
+			'Image to include into the pdf',
+			display_name='Logo / Inline Image'
+		),
+		plugins.ClientOptionString(
+			'link_text',
+			'Text for inserted link',
+			default='Click here to accept',
+			display_name='Link Text'
+		)
+	]
+	req_min_version = '1.7.0b1'
+	req_packages = {
+		'reportlab': has_reportlab
+	}
+	def initialize(self):
+		mailer_tab = self.application.main_tabs['mailer']
+		self.text_insert = mailer_tab.tabs['send_messages'].text_insert
+		self.add_menu_item('Tools > Create PDF Preview', self.make_preview)
+		self.signal_connect('send-precheck', self.signal_send_precheck, gobject=mailer_tab)
+		self.signal_connect('send-target', self.signal_send_target, gobject=mailer_tab)
+		self.signal_connect('send-finished', self.signal_send_finished, gobject=mailer_tab)
+		return True
+
+	def signal_send_precheck(self, _):
+		# file and option checks
+		if not all((self.config['template_file'], self.config['output_pdf'], self.config['link_text'])):
+			self.logger.debug('skipping exporting any attachments due to lack of information provided to run')
+			return 0
+		if not os.path.isfile(self.config['template_file']):
+			self.logger.error('template file does not exist, specify a valid template file')
+			return 0
+		if self.config['logo'] and not os.path.isfile(self.config['template_file']):
+			self.logger.error('specified template or logo file not found')
+		self.logger.debug('pdf template file found, generating attachment')
+		return True
+
+	def make_preview(self, _):
+		if not os.path.isfile(self.config['template_file']):
+			self.logger.error('template file does not exist, please review your options')
+			return 0
+		else:
+			outfile = self.expand_path(self.config['output_pdf'])
+			pdf_file = platypus.SimpleDocTemplate(outfile,
+				pagesize=letter,
+				rightMargin=72,
+				leftMargin=72,
+				topMargin=72,
+				bottomMargin=18
+			)
+			url = self.application.config['mailer.webserver_url']
+			pdf = self.get_template(url)
+			pdf_file.multiBuild(pdf)
+			self.logger.info('created, check ' + self.config['output_pdf'])
+
+	def signal_send_target(self, _, target):
+		outfile = self.expand_path(self.config['output_pdf'])
+		pdf_file = platypus.SimpleDocTemplate(outfile,
+			pagesize=letter,
+			rightMargin=72,
+			leftMargin=72,
+			topMargin=72,
+			bottomMargin=18
+		)
+		url = self.application.config['mailer.webserver_url'] + '?uid=' + target.uid
+		pdf = self.get_template(url)
+		try:
+			pdf_file.multiBuild(pdf)
+			self.attach_pdf(outfile)
+		except Exception as err:
+			self.logger.error('pdf could not be opened: ' + repr(err))
```

Comment:
This is a pretty generic exception handling block. It would be helpful to include the stack trace by using `self.logger.error(..., exc_info=True)`. That way if there's an issue it will be easier to troubleshoot.